controller/src/beerocks/master/wfa_ca.cpp

```diff
@@ -669,3 +724,160 @@ bool wfa_ca::create_cmdu(ieee1905_1::CmduMessageTx &cmdu_tx, ieee1905_1::eMessag
 
     return true;
 }
+
+/**
+ * @brief Parse command "DEV_SEND_1905" TLVs parameters into an ordered list of TLV structure
+ * "tlv_hex_t", and does data validation.
+ * 
+ * @param[in,out] tlv_hex_list An ordered list of TLVs of type "tlv_hex_t".
+ * @param[in] params An unordered map containing parsed parameters of a CAPI command. 
+ * @param[out] err_string Contains an error description if the function fails.
+ * @return true if successful, false if not.
+ */
+bool wfa_ca::get_send_1905_1_tlv_hex_list(std::list<tlv_hex_t> &tlv_hex_list,
+                                          std::unordered_map<std::string, std::string> &params,
+                                          std::string &err_string)
+{
+    static const int max_tlv                          = 256; // Magic number
+    size_t tlv_idx                                    = 0;
+    bool skip                                         = false;
+    bool finish                                       = false;
+    static const std::vector<std::string> tlv_members = {"tlv_type", "tlv_length", "tlv_value"};
+
+    do {
+        tlv_hex_t tlv_hex;
+        for (size_t tlv_member_idx = 0; tlv_member_idx < tlv_members.size(); tlv_member_idx++) {
```

Comment:
There are so many conditions on `tlv_member_idx` inside this loop that I wonder if it's really worth to make a loop for it.

Reply:
I think it is better this way, instead of comparing it to string every time, which will create a larger code.

Reply:
What I meant is to remove the loop. There is not that much that is in common for the three members. In terms of lines of code, just removing the conditions on tlv_member_idx may already reduce the number of lines.